### ISSUE TITLE
Implicit dependencies on bash-compatible shells

### ISSUE THREAD

**Issue by iamvery** [pact-foundation/pact-ruby#156]
I ran into some issues using Pact with https://fishshell.com/. I wasn't able to track them all down, but one thing I noticed is there are some built-in assumptions about how to set environment variables on generated commands, e.g. https://github.com/pact-foundation/pact-ruby/blob/a94d4e12e5481170c92e467e2a20e48f570eb03a/lib/pact/tasks/task_helper.rb#L26. There may be other problems. I actually didn't see this one initially, but the other was more difficult to pin down. Here's part of the output:

```
NoMethodError:
  undefined method `<<' for {:read_timeout=>60, :continue_timeout=>nil, :debug_output=>nil}:Hash
  Did you mean?  <
# ./spec/service_providers/pact_helper.rb:10:in `block (2 levels) in <top (required)>'
# ./spec/service_providers/pact_helper.rb:9:in `block in <top (required)>'
# ./spec/service_providers/pact_helper.rb:8:in `<top (required)>'
# ./spec/acceptance/api/v1/client_spec.rb:5:in `require_relative'
```

Not a huge deal, but I thought it might be helpful to start the conversation. I did a quick search and didn't see anything existing.

Thanks for all your great work on a useful tool!

**Comment by bethesque** [comment]
I remember seeing the << one occasionally, and it's usually due to the version of ruby and the particular gem versions I have installed. I can't remember what I do to fix it now!

If there are small, backwards compatible things we can do to make it run on fish shell, I'm ok with that. Happy to accept PRs. We might be able to set the environment variables using the arguments to the actual shell execution function, rather than setting them in the text.

**Comment by iamvery** [comment]
Good call on the `<<`! I realized this morning that it cropped up for me on a really old version of webmock and Ruby >2.3. Sorry for sending blame pact's way for that 🤦‍♂️ 

Thank you for your quick responses to my questions! If I can find a little more time, I try to send some PRs your way for more general shell support ❤️

**Comment by iamvery** [comment]
I dug in a little more on this, and it seems I was wrong about the environment variable style as well. I couldn't quite pin down the reason, but it may be that Ruby's [`system`](http://ruby-doc.org/core-2.5.0/Kernel.html#method-i-system) method uses /bin/sh rather than the user's shell.

I started to send up a PR that changes the generated command to be prefixed with `env`, but in my testing it proved unnecessary. 

```
» cat lol.rb
puts ENV['lol']
» lol=wat ruby lol.rb
fish: Unsupported use of '='. To run 'ruby' with a modified environment, please use 'env lol=wat ruby…'
» env lol=wat ruby lol.rb
wat
» irb
irb(main):001:0> system 'lol=wat ruby lol.rb'
wat
=> true
```

Sorry to bring up the misplaced issue! Closing it. Will circle back if I do, in fact, run into any real problems ❤️
